includes/data-stores/class-wc-product-data-store-cpt.php

```diff
@@ -1939,8 +1996,7 @@ public function query( $query_vars ) {
 	protected function get_data_for_lookup_table( $id, $table ) {
 		if ( 'wc_product_meta_lookup' === $table ) {
 			$price_meta   = (array) get_post_meta( $id, '_price', false );
-			$manage_stock = get_post_meta( $id, '_manage_stock', true );
-			$stock        = 'yes' === $manage_stock ? wc_stock_amount( get_post_meta( $id, '_stock', true ) ) : null;
+			$stock        = wc_stock_amount( get_post_meta( $id, '_stock', true ) );
```

Comment:
I believe we still need the `_manage_stock` check as, if stock is not managed, we want to set `$stock` to `null` instead of `0`? I haven't checked all the details, but this is probably needed to differentiate products that are out of stock and where the stock is not managed.

Reply:
Haven't seen it used anywhere to take advantage of this difference, but maybe I missed something and it's safer to not change it.
With the introduction of other changes in this PR, it seems like this change is no longer needed, so I put back the manage_stock check.

Reply:
> Haven't seen it used anywhere to take advantage of this difference

Both the report and the dashboard widget that displays out of stock products depend on `null` being used for products that don't have managed stock enabled to work properly. For example, here is the query used in the dashboard widget:

https://github.com/woocommerce/woocommerce/blob/10abe6d4be9351d9bcc86db1afe5264567786221/includes/admin/class-wc-admin-dashboard.php#L232

Reply:
You're right, sorry, I can see that would cause problems now. Thanks for spotting it.